extension/README.md.args.mjs

```diff
@@ -42,110 +36,342 @@ yarn start
+\`\`\`
+
+ >📝 Note: If you use named arguments in your event (e.g. \`event Contribution(address indexed contributor, uint256 amount)\`), you'll need to update \`/packages/nextjs/app/contributions/page.tsx\` to reference event parameters by their names instead of numeric indices.
+
+### Implementing the \`contribute()\` function
+
+> 👩‍💻 Now focus on writing your \`contribute()\` function. The payable method already exists but is empty. Go fill it with logic!
+
+The goal of this function is to allow anyone to contribute to the pool of funds. To do this effectively it will need to do the following:
+- Update the \`balances\` mapping
+- Emit the \`Contribute\` event 
+
+<details markdown='1'>
+<summary>🔎 Hint</summary>
+
+You can set mappings like you would access a Javascript array.
+For a mapping like this \`mapping(address => uint256) public dir\` you would access is like this:
+\`\`\`solidity
+address addr = 0x1234...5678;
+dir[addr] += 1;
+\`\`\`
+You need to use the address for the sender of the transaction and you will need to know how much value was sent. Is there an easy way to access these details about the transaction \`msg\`? 🤔 
+
+Go check https://solidity-by-example.org/ if you need help on the syntax.
+
+<details markdown='1'>
+
+<summary>🎯 Solution</summary>
+
+\`\`\`solidity
+function contribute() public payable {
+	balances[msg.sender] += msg.value;
+	emit Contribution(msg.sender, msg.value);
+}
 \`\`\`
 
-> 👩‍💻 Write your \`stake()\` function and test it with the \`Debug Contracts\` tab in the frontend.
+</details>
+</details>
+
+### Try it out
+
+👩‍💻 Now redeploy (\`yarn deploy\`) and go test your function using the \`Debug Contracts\` tab in the front end.
 
-![debugContracts](https://github.com/scaffold-eth/se-2-challenges/assets/55535804/1a888e31-a79b-49ef-9848-357c5cee445a)
+![debugContracts](https://github.com/user-attachments/assets/7d50245b-72f5-433e-b327-0c7e70e83e51)
 
 > 💸 Need more funds from the faucet? Click on _"Grab funds from faucet"_, or use the Faucet feature at the bottom left of the page to get as much as you need!
 
 ![Faucet](https://github.com/scaffold-eth/se-2-challenges/assets/55535804/e82e3100-20fb-4886-a6bf-4113c3729f53)
 
-> ✏ Need to troubleshoot your code? If you import \`hardhat/console.sol\` to your contract, you can call \`console.log()\` right in your Solidity code. The output will appear in your \`yarn chain\` terminal.
+> ✏ Need to troubleshoot your code? \`hardhat/console.sol\` is already imported in your contract so you can call \`console.log()\` right in your Solidity code. The output will appear in your \`yarn chain\` terminal.
+
+---
 
 ### 🥅 Goals
 
-- [ ] Do you see the balance of the \`Staker\` contract go up when you \`stake()\`?
+- [ ] Do you see the balance of the \`CrowdFund\` contract go up when you \`contribute()\`?
 - [ ] Is your \`balance\` correctly tracked?
-- [ ] Do you see the events in the \`Stake Events\` tab?
+- [ ] Do you see the events in the \`Contributions\` tab?
```

Comment:
getting this (installed challenge with create-eth@latest)

<img width="699" height="60" alt="Image" src="https://github.com/user-attachments/assets/6f269d9d-1fa1-48da-9379-3c860361c59c" />

Need to import `Address` from `scaffold-ui` like on the main page